### PR TITLE
fix: Launcher freeze

### PR DESCRIPTION
**Pull request by GeekCornerGH**
If you were opening the MS Login while in settings and close the window before finishing auth, the launcher throws an error. This patch fixes the issue

### CODE REVIEW

File: app/assets/js/scripts/settings.js [hunk 1]
```diff
@@ -368,13 +368,13 @@ ipcRenderer.on(MSFT_OPCODE.REPLY_LOGIN, (_, ...arguments_) => {
...
             switchView(getCurrentView(), viewOnClose, 500, 500, () => {
                 // TODO Dont know what these errors are. Just show them I guess.
                 // This is probably if you messed up the app registration with Azure.
                 console.log('Error getting authCode, is Azure application registered correctly?')
                 console.log(error)
-                console.log(error_description)
+                console.log(queryMap.error_description)
                 console.log('Full query map', queryMap)
                 let error = queryMap.error // Error might be 'access_denied' ?
                 let errorDesc = queryMap.error_description
```
Comment (dscalzi) [inline app/assets/js/scripts/settings.js:380]: just move the log statement down after the variables are initialized

File: app/assets/js/scripts/settings.js [hunk 2]
```diff
@@ -368,13 +368,13 @@ ipcRenderer.on(MSFT_OPCODE.REPLY_LOGIN, (_, ...arguments_) => {
         const viewOnClose = arguments_[2]
 
         // Error from request to Microsoft.
-        if (Object.prototype.hasOwnProperty.call(queryMap, 'error')) {
+        if (Object.prototype.hasOwnProperty.call(queryMap, error)) {
```
Comment (dscalzi) [inline app/assets/js/scripts/settings.js:371]: ? error is not defined anywhere

**Review by dscalzi — CHANGES_REQUESTED** [review]

File: app/assets/js/scripts/settings.js (hunk 2 above)
Comment (GeekCornerGH) [reply app/assets/js/scripts/settings.js:371]: For some reason it is?

File: app/assets/js/scripts/settings.js (hunk 1 above)
Comment (GeekCornerGH) [reply app/assets/js/scripts/settings.js:380]: Alright

**Review by dscalzi — APPROVED** [review]

**Pull request merged**